engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetFboWriteMask.java

```diff
@@ -83,21 +83,21 @@ public StateChange getDefaultInstance() {
 
     @Override
     public int hashCode() {
-        return Objects.hashCode(fbo.fboId);
+        return Objects.hashCode(fbo);
     }
 
     @Override
     public boolean equals(Object obj) {
         return (obj instanceof SetFboWriteMask)
-                && fbo.fboId == ((SetFboWriteMask) obj).fbo.fboId
+                && fbo == ((SetFboWriteMask) obj).fbo
```

Comment:
I might be (read: I probably am) misunderstanding you, but doesn't my change upon up the door to two FBOs having same ID ? Earlier, id was compared, so two FBOs having same id are the same thing. Now, since actual objects are being compared, two different objects might have same id, but they will still be different.

That being said, I think one fbo object can only be associated with one id and vice versa (strict one-on-one relation), so I simplified it down. What do you recommend I do?